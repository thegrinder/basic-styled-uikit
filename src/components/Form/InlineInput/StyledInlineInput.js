import { bool } from 'prop-types';
import styled, { css } from 'styled-components';
import {
  getInlineInputCommonStyle,
  getInlineInputStyle,
} from '../formSelectors';
import { rem } from '../../Typography';

const propTypes = {
  invalid: bool.isRequired,
  submitting: bool.isRequired,
};

const StyledInlineInput = styled.input`
  -webkit-appearance: none;
  box-sizing: border-box;
  max-width: 100%;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  background-color: transparent;
  transition: 0.2s ease-in-out;
  transition-property: color, background-color, border;
  overflow: visible;

  vertical-align: middle;
  display: inline-block;
  border-radius: ${rem(6)};
  height: ${rem(40)};
  padding: 0 ${rem(10)};
  &:focus {
    outline: none;
  }
  &::placeholder {
    color: inherit;
  }
  ${({ theme, invalid, submitting }) => css`
    ${getInlineInputCommonStyle(theme)}
    ${getInlineInputStyle(theme, invalid, 'normal')}
    ${submitting && `padding-right : ${rem(36)};`}
    &:hover {
      ${getInlineInputStyle(theme, invalid, 'hover')}
    }
    &:focus {
      outline: none;
      ${getInlineInputStyle(theme, invalid, 'active')}
    }
    &:disabled {
      ${getInlineInputStyle(theme, invalid, 'disabled')}
    }
  `}
`;

StyledInlineInput.propTypes = propTypes;

export default StyledInlineInput;
