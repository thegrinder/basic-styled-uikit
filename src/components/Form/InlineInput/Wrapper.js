import styled from 'styled-components';

const Wrapper = styled.div`
  position: relative;
`;

export default Wrapper;
