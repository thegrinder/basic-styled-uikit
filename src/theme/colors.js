import { getColors } from './themeSelectors';

export const getColor = color => ({ theme }) => getColors(theme)[color];

const colors = {
  neutral0: '#fff',
  neutral100: '#f8f9fa',
  neutral400: '#ced4da',
  neutral600: '#868e96',
  neutral1000: '#161616',

  primary400: '#42A5F5',
  primary600: '#1E88E5',
  success400: '#66BB6A',
  success600: '#43A047',
  warning400: '#FFEE58',
  warning600: '#FDD835',
  danger400: '#FF8A65',
  danger600: '#F4511E',
};

export default colors;
