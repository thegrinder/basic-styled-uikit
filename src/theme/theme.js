import button from '../components/Button/buttonTheme';
import link from '../components/Link/linkTheme';
import form from '../components/Form/formTheme';
import text from '../components/Text/textTheme';
import heading from '../components/Headings/headingTheme';
import colorBox from '../components/ColorBox/colorBoxTheme';

const theme = {
  uiKit: {
    button,
    link,
    form,
    text,
    heading,
    colorBox,
  },
};

export default theme;
