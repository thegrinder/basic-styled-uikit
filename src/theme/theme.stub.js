import { rem } from 'basic-styled-uikit';

const colors = {
  transparent: 'transparent',
  white: '#fff',
  gray50: '#fafafa',
  gray100: '#f8f9fa',
  gray200: '#e9ecef',
  gray300: '#dee2e6',
  gray400: '#ced4da',
  gray500: '#adb5bd',
  gray600: '#868e96',
  gray700: '#495057',
  gray800: '#343a40',
  gray900: '#212529',
  black: '#161616',

  primary50: '#e3f2fd',
  primary100: '#bbdefb',
  primary200: '#90caf9',
  primary300: '#64b5f6',
  primary400: '#42a5f5',
  primary500: '#2196f3',
  primary600: '#1e88e5',
  primary700: '#1976d2',
  primary800: '#1565c0',
  primary900: '#0d47a1',

  success50: '#e8f5e9',
  success100: '#c8e6c9',
  success200: '#a5d6a7',
  success300: '#81c784',
  success400: '#66bb6a',
  success500: '#4caf50',
  success600: '#43a047',
  success700: '#388e3c',
  success800: '#2e7d32',
  success900: '#1b5e20',

  warning50: '#fffde7',
  warning100: '#fff9c4',
  warning200: '#fff59d',
  warning300: '#fff176',
  warning400: '#ffee58',
  warning500: '#ffeb3b',
  warning600: '#fdd835',
  warning700: '#fbc02d',
  warning800: '#f9a825',
  warning900: '#f57f17',

  danger50: '#ffebee',
  danger100: '#ffcdd2',
  danger200: '#ef9a9a',
  danger300: '#e57373',
  danger400: '#ef5350',
  danger500: '#f44336',
  danger600: '#e53935',
  danger700: '#d32f2f',
  danger800: '#c62828',
  danger900: '#b71c1c',
};

const theme = {
  uiKit: {
    colors,
    // button theme
    button: {
      btnTypes: {
        basic: {
          normal: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray300,
          },
          hover: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray400,
          },
          active: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray500,
          },
        },
        default: {
          normal: {
            color: colors.black,
            backgroundColor: colors.gray300,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.black,
            backgroundColor: colors.gray400,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.black,
            backgroundColor: colors.gray500,
            borderColor: colors.transparent,
          },
        },
        primary: {
          normal: {
            color: colors.white,
            backgroundColor: colors.primary500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.primary600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.primary700,
            borderColor: colors.transparent,
          },
        },
        danger: {
          normal: {
            color: colors.white,
            backgroundColor: colors.danger500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.danger600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.danger700,
            borderColor: colors.transparent,
          },
        },
      },
      btnSizes: {
        small: {
          padding: `0 ${rem(15)}`,
          lineHeight: rem(28),
          fontSize: rem(14),
        },
        default: {
          padding: `0 ${rem(30)}`,
          fontSize: rem(16),
          lineHeight: rem(38),
        },
        large: {
          padding: `0 ${rem(40)}`,
          lineHeight: rem(53),
          fontSize: rem(22),
        },
      },
      common: {
        fontWeight: '700',
        letterSpacing: '.4px',
      },
    },
    // form theme
    form: {
      regular: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.white,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.white,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.white,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      inline: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.transparent,
              borderColor: colors.transparent,
              color: colors.gray700,
            },
            hover: {
              backgroundColor: colors.transparent,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.transparent,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.transparent,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      // radio and checkbox theme
      misc: {
        states: {
          valid: {
            normal: {
              borderColor: colors.gray300,
              backgroundColor: colors.transparent,
            },
            active: {
              borderColor: colors.primary500,
              backgroundColor: colors.transparent,
            },
            checked: {
              borderColor: colors.transparent,
              backgroundColor: colors.primary500,
            },
          },
          invalid: {
            borderColor: colors.danger500,
            backgroundColor: colors.transparent,
          },
        },
        common: {},
      },
      // toggle theme
      toggle: {
        normal: {
          backgroundColor: colors.gray400,
        },
        checked: {
          backgroundColor: colors.primary500,
        },
      },
    },
    text: {
      sizings: {
        small: rem(14),
        medium: rem(16),
        large: rem(20),
      },
    },
    heading: {
      sizings: {
        h1: rem(36),
        h2: rem(32),
        h3: rem(28),
        h4: rem(24),
        h5: rem(20),
        h6: rem(18),
      },
    },
  },
};

export default theme;
