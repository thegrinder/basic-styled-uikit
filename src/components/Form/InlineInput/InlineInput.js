import React from 'react';
import { bool, node } from 'prop-types';
import styled from 'styled-components';
import { rem } from '../../Typography';
import StyledInlineInput from './StyledInlineInput';

const Wrapper = styled.div`
  position: relative;
`;

const SpinnerWrapper = styled.div`
  position: absolute;
  right: ${rem(8)};
  top: 0;
  bottom: 0;
  display: flex;
  align-items: center;
`;

const propTypes = {
  /** submitting flag that shows a component passed to renderSpinner prop */
  submitting: bool,
  /** disabled flag */
  disabled: bool,
  /** invalid flag */
  invalid: bool,
  /** spinner component shown when submitting flag is true */
  renderSpinner: node,
};

const defaultProps = {
  submitting: false,
  disabled: false,
  invalid: false,
};

const InlineInput = ({
  submitting,
  disabled,
  invalid,
  renderSpinner,
  ...rest
}) => (
  <Wrapper>
    <StyledInlineInput
      submitting={submitting}
      disabled={submitting || disabled}
      invalid={invalid}
      {...rest}
    />
    {submitting && <SpinnerWrapper>{renderSpinner}</SpinnerWrapper>}
  </Wrapper>
);

InlineInput.propTypes = propTypes;
InlineInput.defaultProps = defaultProps;

export default InlineInput;
