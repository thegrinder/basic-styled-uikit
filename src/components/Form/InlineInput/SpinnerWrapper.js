import styled from 'styled-components';
import { rem } from '../../../theme/typography';

const SpinnerWrapper = styled.div`
  position: absolute;
  right: ${({ theme }) => rem(theme, 8)};
  top: 0;
  bottom: 0;
  display: flex;
  align-items: center;
`;

export default SpinnerWrapper;
