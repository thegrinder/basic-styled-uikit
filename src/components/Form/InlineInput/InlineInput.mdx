---
name: InlineInput
menu: Components
---

import { Playground, Props } from 'docz';
import InlineInput from './InlineInput';
import { InputWrapper, Spinner } from '../../../../docz';

# InlineInput

<Props of={InlineInput} />

### InlineInput

<Playground>
  <InputWrapper>
    <InlineInput placeholder="Hover here" />
  </InputWrapper>
</Playground>

### Invalid InlineInput

<Playground>
  <InputWrapper>
    <InlineInput invalid={true} />
  </InputWrapper>
</Playground>

### Disabled InlineInput

<Playground>
  <InputWrapper>
    <InlineInput disabled={true} />
  </InputWrapper>
</Playground>

### Disabled InlineInput with a spinner

<Playground>
  <InputWrapper>
    <InlineInput
      disabled={true}
      submitting={true}
      renderSpinner={<Spinner color="#ccc" />}
    />
  </InputWrapper>
</Playground>
