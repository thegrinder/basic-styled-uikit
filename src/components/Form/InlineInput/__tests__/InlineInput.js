import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, cleanup } from '@testing-library/react';


import theme from '../../../../theme/theme';
import InlineInput from '../InlineInput';


const renderComponent = (props = {}) => render(
  <ThemeProvider theme={theme}>
    <InlineInput {...props} />
  </ThemeProvider>,
);

describe('<InlineInput />', () => {
  afterEach(cleanup);

  it('should render correctly with default props', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should render correctly with custom props', () => {
    const spinner = <span data-testid="spinner" />;
    const { container: { firstChild }, getByTestId } = renderComponent({
      invalid: false,
      submitting: true,
      renderSpinner: spinner,
    });
    const spinnerElement = getByTestId('spinner');
    expect(firstChild).toBeDefined();
    expect(firstChild).toContainElement(spinnerElement);
  });
});
