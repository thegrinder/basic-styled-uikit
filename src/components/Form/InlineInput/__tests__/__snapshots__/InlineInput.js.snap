// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineInput /> should render correctly with default props 1`] = `
<div
  class="Wrapper-sc-1e6m9j8-0 gUPYuR"
>
  <input
    class="StyledInlineInput-sc-5g60q6-0 YXTKn"
  />
</div>
`;
